### PR TITLE
Use T.workspace for compatibility with Mill 0.12

### PR DESCRIPTION
**Pull request by lolgab**
Dropped support for Mill 0.9 which does not support `T.workspace`.

### CODE REVIEW

**Comment by Gedochao** [comment]
cc @alexarchambault

File: plugin/src/io/github/alexarchambault/millnativeimage/NativeImage.scala [hunk 1]
```diff
@@ -316,10 +317,7 @@ object NativeImage {
   }
 
   def vcvarsOpt: Option[os.Path] = {
-    val workspace = sys.env.get("MILL_WORKSPACE_ROOT") match {
-      case Some(dir) => os.Path(dir)
-      case None => os.pwd
-    }
+    val workspace = WorkspaceRoot.workspaceRoot.getOrElse(os.pwd)
```
Comment (lolgab) [inline plugin/src/io/github/alexarchambault/millnativeimage/NativeImage.scala]: This doesn't work on Mill `0.10`. We could use it with version-specific sources.
Let me know what you prefer and I can implement it.

Comment (alexarchambault) [reply plugin/src/io/github/alexarchambault/millnativeimage/NativeImage.scala]: I reverted it, version-specific sources would be an overkill here, I think

**Review by alexarchambault — APPROVED** [review]
Thanks @lolgab!

**Pull request merged**